File: .py
class Square:
    def __init__(self, side):
        self.side = side

    def getPeri(self):
        return 4 * self.side

    def getArea(self):
        return self.side ** 2


class Cube(Square):
    def getVolume(self):
        return self.side ** 3


c = Cube(4)
print("Area:", c.getArea())
print("Volume:", c.getVolume())

File: .cpp
#include <iostream>
using namespace std;

class Square {
protected:
    double side;

public:
    Square(double s) { side = s; }
    double getPeri() { return 4 * side; }
    double getArea() { return side * side; }
};

class Cube : public Square {
public:
    Cube(double s) : Square(s) {}
    double getVolume() { return side * side * side; }
};

int main() {
    Cube c(4);
    cout << "Area: " << c.getArea() << endl;
    cout << "Volume: " << c.getVolume();
    return 0;
}
